### PR TITLE
docs: add missing libboost dependencies to build instructions

### PR DESCRIPTION
**Pull request by flotwig**
These dependencies seem to be required to build Cemu on Ubuntu 23.10 and Ubuntu 22.04 but are missing from `BUILD.md`. See https://github.com/cemu-project/Cemu/issues/223#issuecomment-1774122908 for more context.

Validated that these packages exist as far back as 22.04.1 LTS and that installing them fixes the error in #223.

### CODE REVIEW

**Comment by qurious-pixel** [comment]
Why not include libboost-filesystem-dev as well, and drop the version requirement. 

https://github.com/cemu-project/Cemu/blob/f3c95f72e74d8a5f5873061fbb994643c63ec9c5/CMakeLists.txt#L130

File: BUILD.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ To compile Cemu, a recent enough compiler and STL with C++20 support is required
 ### Installing dependencies
 
 #### For Ubuntu and derivatives:
-`sudo apt install -y cmake curl clang-15 freeglut3-dev git libgcrypt20-dev libglm-dev libgtk-3-dev libpulse-dev libsecret-1-dev libsystemd-dev nasm ninja-build`
+`sudo apt install -y cmake curl clang-15 freeglut3-dev git libgcrypt20-dev libglm-dev libgtk-3-dev libpulse-dev libsecret-1-dev libsystemd-dev nasm ninja-build libboost-program-options1.74-dev libboost-nowide1.74-dev`
```
Comment (qurious-pixel) [inline BUILD.md:27]: `libboost-program-options-dev libboost-nowide-dev libboost-filesystem-dev`

Comment (flotwig) [reply BUILD.md:27]: :+1: I already had libboost-filesystem-dev so I didn't notice it was missing as well. Thanks.

Comment (flotwig) [reply BUILD.md:27]:
```suggestion
`sudo apt install -y cmake curl clang-15 freeglut3-dev git libgcrypt20-dev libglm-dev libgtk-3-dev libpulse-dev libsecret-1-dev libsystemd-dev nasm ninja-build libboost-program-options-dev libboost-nowide-dev libboost-filesystem-dev`
```

**Comment by Exzap** [comment]
Are you building with `-DENABLE_VCPKG=OFF`? Normally vcpkg should take care of boost.

**Comment by flotwig** [comment]
> Are you building with `-DENABLE_VCPKG=OFF`? Normally vcpkg should take care of boost.

@Exzap I am using `cmake -S . -B build -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/clang-15 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-15 -G Ninja -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja`, just the command recommended in the doc.

**Comment by qurious-pixel** [comment]
The CI is successfully building on Ubuntu-22.04.
The steps that are missing from Issue https://github.com/cemu-project/Cemu/issues/223 are the the following:
```bash
sudo apt install -y git curl cmake ninja-build nasm libgtk-3-dev libsecret-1-dev libgcrypt20-dev libsystemd-dev freeglut3-dev libpulse-dev
git clone --recursive https://github.com/cemu-project/Cemu
cd Cemu
cd dependencies/vcpkg
git fetch --unshallow
git pull --all
cd -
./dependencies/vcpkg/bootstrap-vcpkg.sh
cmake -S . -B build -DCMAKE_BUILD_TYPE=release -DPORTABLE=OFF -DCMAKE_C_COMPILER=/usr/bin/clang-15 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-15 -G Ninja -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja
cmake --build build
```

**Comment by Exzap** [comment]
I am closing this because it is more of a workaround than a proper fix for whatever the problem with the boost vcpkg port is (it shouldn't be necessary to install system boost). We also recently updated vcpkg and the boost dependency, so maybe this has been resolved anyway?

**Comment by flotwig** [comment]
@Exzap I'm not sure if this is fixed, it seems like a new error is happening earlier in the build process now. I uninstalled the added packages from my Ubuntu 23.10 system. With the cmake/clang instructions:

```
➜  cemu.test git:(main) cmake -S . -B build -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -G Ninja
cmake --build build
-- Running vcpkg install
error: failed to execute: /usr/bin/git --git-dir=/tmp/cemu.test/dependencies/vcpkg/.git --work-tree=/tmp/cemu.test/dependencies/vcpkg/buildtrees/versioning_/versions/boost-modular-build-helper/843c4331dc1453c80f99e25be4513c99074c5cf2_81035.tmp -c core.autocrlf=false read-tree -m -u 843c4331dc1453c80f99e25be4513c99074c5cf2
vcpkg was cloned as a shallow repository in: /tmp/cemu.test/dependencies/vcpkg/.git
Try again with a full vcpkg clone.
error: git failed with exit code: (128).
fatal: failed to unpack tree object 843c4331dc1453c80f99e25be4513c99074c5cf2
note: while checking out port boost-modular-build-helper with git tree 843c4331dc1453c80f99e25be4513c99074c5cf2
-- Running vcpkg install - failed
CMake Error at dependencies/vcpkg/scripts/buildsystems/vcpkg.cmake:899 (message):
  vcpkg install failed.  See logs for more information:
  /tmp/cemu.test/build/vcpkg-manifest-install.log
Call Stack (most recent call first):
  /usr/share/cmake-3.27/Modules/CMakeDetermineSystem.cmake:148 (include)
  CMakeLists.txt:36 (project)


CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-- Configuring incomplete, errors occurred!
No such file or directory
CMake Error: Generator: execution of make failed. Make command was: 
```

With the gcc instructions:

```
➜  cemu.test git:(main) cmake -S . -B build -DCMAKE_BUILD_TYPE=release -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++ -G Ninja
cmake --build build

-- Running vcpkg install
error: failed to execute: /usr/bin/git --git-dir=/tmp/cemu.test/dependencies/vcpkg/.git --work-tree=/tmp/cemu.test/dependencies/vcpkg/buildtrees/versioning_/versions/boost-modular-build-helper/843c4331dc1453c80f99e25be4513c99074c5cf2_81182.tmp -c core.autocrlf=false read-tree -m -u 843c4331dc1453c80f99e25be4513c99074c5cf2
vcpkg was cloned as a shallow repository in: /tmp/cemu.test/dependencies/vcpkg/.git
Try again with a full vcpkg clone.
error: git failed with exit code: (128).
fatal: failed to unpack tree object 843c4331dc1453c80f99e25be4513c99074c5cf2
note: while checking out port boost-modular-build-helper with git tree 843c4331dc1453c80f99e25be4513c99074c5cf2
-- Running vcpkg install - failed
CMake Error at dependencies/vcpkg/scripts/buildsystems/vcpkg.cmake:899 (message):
  vcpkg install failed.  See logs for more information:
  /tmp/cemu.test/build/vcpkg-manifest-install.log
Call Stack (most recent call first):
  /usr/share/cmake-3.27/Modules/CMakeDetermineSystem.cmake:148 (include)
  CMakeLists.txt:36 (project)


CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-- Configuring incomplete, errors occurred!
No such file or directory
CMake Error: Generator: execution of make failed. Make command was: 
```

**Pull request closed without merging**